src/Security/Voter/UserRequestVoter.php

```diff
@@ -17,30 +16,41 @@ public function __construct(private readonly AuthorizationCheckerInterface $chec
 
     protected function supports(string $attribute, mixed $subject): bool
     {
-        return in_array($attribute, [Permissions::UPDATE, Permissions::READ]) && $subject instanceof UserRequest;
+        return in_array($attribute, [Permissions::UPDATE, Permissions::READ, Permissions::DELETE]) && $subject instanceof UserRequest;
     }
 
     protected function voteOnAttribute(string $attribute, mixed $subject, TokenInterface $token): bool
     {
+        /** @var User $user */
         $user = $token->getUser();
-        if (!$user instanceof UserInterface || !$subject instanceof UserRequest) {
+        if (!$user instanceof User || !$subject instanceof UserRequest) {
             return false;
         }
 
         if ($this->checker->isGranted('ROLE_TECH_TEAM')) {
             return true;
         }
 
-        if (Permissions::UPDATE === $attribute && $user !== $subject->getUser()) {
-            return false;
-        }
+        return match ($attribute) {
+            Permissions::READ => $this->canRead($subject, $user),
+            Permissions::UPDATE => $this->canUpdate($subject, $user),
+            Permissions::DELETE => $this->canDelete($subject, $user),
+            default => false,
+        };
+    }
 
-        if (Permissions::READ === $attribute) {
-            return true;
-        } elseif (Permissions::UPDATE === $attribute) {
-            return $subject->isDraft();
-        }
+    private function canRead(UserRequest $subject, User $user): bool
+    {
+        return $this->checker->isGranted('ROLE_TEAM') || $user === $subject->getUser();
+    }
 
-        return false;
+    private function canUpdate(UserRequest $subject, User $user): bool
+    {
+        return $subject->isDraft() && $user === $subject->getUser();
+    }
+
+    private function canDelete(UserRequest $subject, User $user): bool
+    {
+        return $user === $subject->getUser();
```

Comment:
Should we not return `$this->canRead()`

Reply:
Or something else, but I like to cascade permissions

Reply:
Good idea, we can return `$this->canRead() && $user === $subject->getUser()` 
Just `$this->canRead()` would be incorrect because a Team user can see all userRequests but can only delete the one he owns.